integration/shareddata/shareddata.go

```diff
@@ -83,6 +84,32 @@ func AllProviders() []Provider {
 	return maps.Values(res)
 }
 
+// Providers are array of providers.
+type Providers []Provider
+
+// Remove returns an array of providers without the specified provider.
+func (ps Providers) Remove(provider string) Providers {
```

Comment:
I like the idea of this function! With it, we can keep `AllProviders` in as many places as possible and exclude only those that are not supported because of a specific reason.

Reply:
Maybe also the signature could be `Remove(provider ...string)`, so if a few providers are removed, we don't need to call `Remove` twice.

Reply:
Good idea :+1: Applied this